File: example/javalib/testing/4-test-grouping/foo/testng/src/foo/HelloTests.java
package foo;

import static org.testng.Assert.assertTrue;
import org.testng.annotations.Test;

public class HelloTests {

  @Test
  public void hello() throws Exception {
    System.out.println("Testing Hello");
    String result = new Foo().hello();
    assertTrue(result.startsWith("Hello"));
    Thread.sleep(1000);
    System.out.println("Testing Hello Completed");
  }
}

File: example/scalalib/testing/5-test-grouping/foo/test/src/HelloTests.scala
package foo
import utest.*
object HelloTests extends TestSuite {
  def tests = Tests {
    test("hello") {
      println("Testing Hello")
      val result = Foo.hello()
      assert(result.startsWith("Hello"))
      Thread.sleep(1000)
      println("Testing Hello Completed")
      result
    }
  }
}
